tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -247,7 +247,16 @@ internal static void SetTagObject(Span span, string key, object? value)
                     span.SetMetric(key, us);
                     break;
                 case int i: // TODO: Can't get here from OTEL API, test with Activity API
```

Comment:
If we can't get here from the otel API, do we need to handle it another way?

Reply:
So we get here from the Activity API - we do test it the comments are outdated

Reply:
Oh wait, we 100% can get here with the OpenTelemetry API as that is what I did